Sources/SpeziHealthKit/HealthKit.swift

```diff
@@ -143,8 +143,11 @@ public final class HealthKit: Module, EnvironmentAccessible, DefaultInitializabl
         guard !authorized else {
             return
         }
-        
-        try await healthStore.requestAuthorization(toShare: [], read: healthKitSampleTypes)
+        let toShareSampleTypes: Set<HKSampleType> = [
+            HKObjectType.quantityType(forIdentifier: .stepCount)!,
+            HKObjectType.quantityType(forIdentifier: .activeEnergyBurned)!
+        ]
+        try await healthStore.requestAuthorization(toShare: toShareSampleTypes, read: healthKitSampleTypes)
```

Comment:
We cannot hard code the sample types here – this will request authorization to share step count and active energy burned in all applications that use SpeziHealthKit. We should figure out a way to add this in the UI test application, I think you should just be able to call 
`healthStore.requestAuthorization(toShare: toShareSampleTypes, read: healthKitSampleTypes)` 
again and it will ask for the updated share permissions

Reply:
The health store is unfortunately a private variable inside of the healthkit module, so I can't access it directly to call `requestAuthorization` and pass in the types to share

Reply:
This might be a bit of a hack, but are you able to just initialize and access your own `HKHealthStore` object from within the UI test app? I just don't think adding additional permissions to the entire SpeziHealthKit package is feasible. The alternative would be to add functionality to SpeziHealthKit that allows you to declare types you want to share (in addition to read permissions)

Reply:
I can't access `healthKitSampleTypes` either since its also a private var, should I just make a separate var for those as well (since we know the 5 types the test app asks to read) alongside `toShareSampleTypes` ?

Reply:
@PSchmiedmayer  Hi Paul, right now we're running into issues in the test app with getting write permissions to inject step count data for testing purposes. I have the button that injects data, but has to do so with a separate instance of the apple HealthStore, since SpeziHealthKit currently doesn't support passing in types to request write access (only read).

 The issue with this is that now the health authorization sheet pops up when triggering manual data collection in the test app, which shouldn't be the case, but this separate instance is necessary because of the private protection levels in the HealthKit module for the sample types passed in to `healthStore.requestAuthorization()`. 
 
I had originally modified SpeziHealthKit directly (unknowingly) but pivoted (as @mjoerke suggested above) and attempted this separate HKStore instance to try to get around it. With the modifications to SpeziHealthKit (hardcoded but it showed that it could work), the tests successfully passed regarding bulkupload, but we obviously can't leave those hardcoded write permissions in. 
 
Recognizing the time crunch we're on with starting internal user studies this week, we wanted some guidance on how to go about this since it feels like we're spending time "hacking" the test app. Given the time crunch with internal user studies starting this week, we felt it might be best to start integrating within the Prisma app. Eventually it might be good to have SpeziHealthKit allow for some write permissions as well so that we can use them for the test app, but we did want to get moving on integrating within Prisma. 

Would love your thoughts/input on this! 🙏🏽

Reply:
@bryant-jimenez I would suggest to manually request permissions to get write permissions when you press the button to inject the data. You can do all the operations in an async function based on the button press (e.g. how to get write permissions: https://developer.apple.com/documentation/healthkit/authorizing_access_to_health_data) and within your UI test use the same UI testing code that we use to handle and dismiss the HealthKit permissions sheet ... that might be the cleanest way to handle this 🚀